rpc/service/src/service.rs

```diff
@@ -522,6 +523,65 @@ impl RpcApi for RpcCoreService {
         Ok(GetCoinSupplyResponse::new(MAX_SOMPI, circulating_sompi))
     }
 
+    async fn get_daa_score_timestamp_estimate_call(
+        &self,
+        request: GetDaaScoreTimestampEstimateRequest,
+    ) -> RpcResult<GetDaaScoreTimestampEstimateResponse> {
+        let session = self.consensus_manager.consensus().session().await;
+        let mut headers = session.async_get_chain_block_samples().await;
+        let mut requested_daa_scores = request.daa_scores.clone();
+        let mut daa_score_timestamp_map = HashMap::<u64, u64>::new();
+
+        headers.reverse();
+        requested_daa_scores.sort_by(|a, b| b.cmp(a));
+
+        let mut header_idx = 0;
+        let mut req_idx = 0;
+
+        // Loop runs at O(n + m) where n = # pp headers, m = # requested daa_scores
+        // Loop will always end because in the worst case the last header with daa_score = 0 (the genesis)
+        // will cause every remaining requested daa_score to be "found in range"
+        while header_idx < headers.len() && req_idx < request.daa_scores.len() {
+            let header = headers.get(header_idx).unwrap();
+            let curr_daa_score = requested_daa_scores[req_idx];
+
+            // Found daa_score in range
+            if header.daa_score <= curr_daa_score {
+                // For the last header, we estimate excess time as seconds
+                let time_adjustment = if header_idx == 0 {
+                    // Assume difference in daa_score corresponds to seconds since the basis header
+                    (curr_daa_score - header.daa_score).checked_mul(1000).unwrap_or(u64::MAX)
+                } else {
+                    // "next" header is the one that we processed last iteration
+                    let next_header = headers.get(header_idx - 1).unwrap();
```

Comment:
use index(square brackets) instead of get + unwrap

Reply:
Addressed in [d405780](https://github.com/kaspanet/rusty-kaspa/pull/268/commits/d405780e503c768a5232cd2d8c43aa049d048e41)